sympy/core/tests/test_subs.py

```diff
@@ -463,6 +463,8 @@ def test_add():
     ans = (-x*(x) - y*(-x)).expand()
     assert e.subs(-y + 1, x) == ans
 
+    #Test issue 18747
```

Comment:
* This is a manifestation of the issue which might change depending on the ordering of the argument. I would prefer a direct tests of the `oo` and `AccumBounds(-1,1)` as shown on the issue and in the OP.

* I was nervous about using anything but the `__add__` method since I had bad experiences with recursion errors when working on `flatten` some years ago. Once burned, twice shy... But if it works then the following could be changed, too:
```
             elif isinstance(o, AccumBounds):
-                coeff = o.__add__(coeff)
+                coeff += o
                 continue
```

* Also, the test that failed the first time around might have something to do with the fact that 0 could be obtained as a "pos_random_int" and would results in a `zoo` being generated instead of a Rational (and I confirmed that that will cause the test to fail). The following diff will avoid that issue:
```diff
diff --git a/sympy/tensor/tests/test_tensor_operators.py b/sympy/tensor/tests/test_tensor_operators.py
index 2fa1101..0faed0d 100644
--- a/sympy/tensor/tests/test_tensor_operators.py
+++ b/sympy/tensor/tests/test_tensor_operators.py
@@ -191,9 +191,9 @@ def test_expand_partial_derivative_sum_rule():


 def test_expand_partial_derivative_constant_factor_rule():
-    pos_random_int1 = sympify(randint(0, 1000))
-    pos_random_int2 = sympify(randint(0, 1000))
-    neg_random_int = sympify(randint(-1000, -1))
+    pos_random_int1 = randint(1, 1000)
+    pos_random_int2 = randint(1, 1000)
+    neg_random_int = -randint(1, 1000)

     c1 = Rational(pos_random_int1, pos_random_int2)
     c2 = Rational(neg_random_int, pos_random_int2)
```